github/event.go

```diff
@@ -27,139 +27,16 @@ func (e Event) String() string {
-		payload = &InstallationTargetEvent{}
-	case "IssueCommentEvent":
-		payload = &IssueCommentEvent{}
-	case "IssuesEvent":
-		payload = &IssuesEvent{}
-	case "LabelEvent":
-		payload = &LabelEvent{}
-	case "MarketplacePurchaseEvent":
-		payload = &MarketplacePurchaseEvent{}
-	case "MemberEvent":
-		payload = &MemberEvent{}
-	case "MembershipEvent":
-		payload = &MembershipEvent{}
-	case "MergeGroupEvent":
-		payload = &MergeGroupEvent{}
-	case "MetaEvent":
-		payload = &MetaEvent{}
-	case "MilestoneEvent":
-		payload = &MilestoneEvent{}
-	case "OrganizationEvent":
-		payload = &OrganizationEvent{}
-	case "OrgBlockEvent":
-		payload = &OrgBlockEvent{}
-	case "PackageEvent":
-		payload = &PackageEvent{}
-	case "PageBuildEvent":
-		payload = &PageBuildEvent{}
-	case "PersonalAccessTokenRequestEvent":
-		payload = &PersonalAccessTokenRequestEvent{}
-	case "PingEvent":
-		payload = &PingEvent{}
-	case "ProjectEvent":
-		payload = &ProjectEvent{}
-	case "ProjectCardEvent":
-		payload = &ProjectCardEvent{}
-	case "ProjectColumnEvent":
-		payload = &ProjectColumnEvent{}
-	case "ProjectV2Event":
-		payload = &ProjectV2Event{}
-	case "ProjectV2ItemEvent":
-		payload = &ProjectV2ItemEvent{}
-	case "PublicEvent":
-		payload = &PublicEvent{}
-	case "PullRequestEvent":
-		payload = &PullRequestEvent{}
-	case "PullRequestReviewEvent":
-		payload = &PullRequestReviewEvent{}
-	case "PullRequestReviewCommentEvent":
-		payload = &PullRequestReviewCommentEvent{}
-	case "PullRequestReviewThreadEvent":
-		payload = &PullRequestReviewThreadEvent{}
-	case "PullRequestTargetEvent":
-		payload = &PullRequestTargetEvent{}
-	case "PushEvent":
-		payload = &PushEvent{}
-	case "ReleaseEvent":
-		payload = &ReleaseEvent{}
-	case "RepositoryEvent":
-		payload = &RepositoryEvent{}
-	case "RepositoryDispatchEvent":
-		payload = &RepositoryDispatchEvent{}
-	case "RepositoryImportEvent":
-		payload = &RepositoryImportEvent{}
-	case "RepositoryVulnerabilityAlertEvent":
-		payload = &RepositoryVulnerabilityAlertEvent{}
-	case "SecretScanningAlertEvent":
-		payload = &SecretScanningAlertEvent{}
-	case "SecurityAdvisoryEvent":
-		payload = &SecurityAdvisoryEvent{}
-	case "SecurityAndAnalysisEvent":
-		payload = &SecurityAndAnalysisEvent{}
-	case "StarEvent":
-		payload = &StarEvent{}
-	case "StatusEvent":
-		payload = &StatusEvent{}
-	case "TeamEvent":
-		payload = &TeamEvent{}
-	case "TeamAddEvent":
-		payload = &TeamAddEvent{}
-	case "UserEvent":
-		payload = &UserEvent{}
-	case "WatchEvent":
-		payload = &WatchEvent{}
-	case "WorkflowDispatchEvent":
-		payload = &WorkflowDispatchEvent{}
-	case "WorkflowJobEvent":
-		payload = &WorkflowJobEvent{}
-	case "WorkflowRunEvent":
-		payload = &WorkflowRunEvent{}
+func (e *Event) ParsePayload() (interface{}, error) {
+	// It would be nice if e.Type were the snake_case name of the event,
+	// but the existing interface uses the struct name instead.
+	var payload interface{} = EventForType(typeToMessageMapping[e.GetType()])
```

Comment:
```suggestion
	payload := EventForType(typeToMessageMapping[e.GetType()])
```

Reply:
Done.  Earlier, I'd had some trouble with the compiler getting the types correct, but I think that was with the map-of-structs rather than map-of-pointers-to-structs solution.